src/Exceptions/ParcelTrapDriverException.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace ParcelTrap\Exceptions;
+
+use Exception;
+use ParcelTrap\Contracts\Driver;
+
+abstract class ParcelTrapDriverException extends Exception
+{
+    private Driver $driver;
```

Comment:
I'm not entirely sure if this will work, but any chance (as this is only set in the `__construct()` method for classes extending this, that this could be readonly and public without the getter/setter? 🤔
```suggestion
    public readonly Driver $driver;
```

Reply:
Funny, cause I had it that way -- the `ApiLimitReachedException` and `ApiAuthenticationFailedException` both had `private readonly Driver $driver` and the `ParcelTrapDriverException` had `getDriver(): Driver` -- however phpstan had 3 issues for this:

ApiLimitReachedException -- $driver written but never used
ApiAuthenticationFailedException -- $driver written but never used
ParcelTrapDriverException -- $driver read but never written

Guess I can revert it to how it was, and add phpstan-ignore lines

Reply:
Yeah, I'm happy to look into why. I'd just use `public readonly Driver $driver` on the abstract (this one), and then reference it in the constructor for th eothers.

Reply:
Without the getters it reduces the phpstan issues to 1... have ignored the line